service/dynamodb/expression/condition.go

```diff
@@ -32,23 +49,150 @@ func Equal(left, right OperandBuilder) ConditionBuilder {
 }
 
 // Equal will create a ConditionBuilder. This will be the method for PathBuilder
+//
+// Example:
+//
+// The following produce equivalent conditions:
+// condition := expression.Equal(expression.NewPath("foo"), expression.NewValue(5))
+// condition := expression.NewPath("foo").Equal(expression.NewValue(5))
 func (p PathBuilder) Equal(right OperandBuilder) ConditionBuilder {
 	return Equal(p, right)
 }
 
 // Equal will create a ConditionBuilder. This will be the method for
 // ValueBuilder
+//
+// Example:
+//
+// The following produce equivalent conditions:
+// condition := expression.Equal(expression.NewValue(10), expression.NewValue(5))
+// condition := expression.NewValue(10).Equal(expression.NewValue(5))
 func (v ValueBuilder) Equal(right OperandBuilder) ConditionBuilder {
 	return Equal(v, right)
 }
 
 // Equal will create a ConditionBuilder. This will be the method for SizeBuilder
+//
+// Example:
+//
+// The following produce equivalent conditions:
+// condition := expression.Equal(expression.NewPath("foo").Size(), expression.NewValue(5))
+// condition := expression.NewPath("foo").Size().Equal(expression.NewValue(5))
 func (s SizeBuilder) Equal(right OperandBuilder) ConditionBuilder {
 	return Equal(s, right)
 }
 
+// And will create a ConditionBuilder with more than two other Conditions as
+// children, representing logical statements that will be logically ANDed
+// together. The resulting ConditionBuilder can be used to build other
+// Conditions or to create an Expression to be used in an operation input. This
+// will be the function call.
+//
+// Example:
+//
+// condition1 := expression.Equal(expression.NewPath("foo"), expression.NewValue(5))
+// condition2 := expression.Less(expression.NewPath("bar"), expression.NewValue(2010))
+// condition3 := expression.NewPath("baz").Between(expression.NewValue(2), expression.NewValue(10))
+// andCondition := expression.And(condition1, condition2, condition3)
+//
+// anotherCondition := expression.Not(andCondition)		// Used in another condition
+// expression, err := andCondition.BuildExpression()	// Used to make an Expression
+func And(cond ...ConditionBuilder) ConditionBuilder {
```

Comment:
This function might be better as `And(left, right ConditionBuilder,  others ...ConditionBuilder)` Since And requires left and right hand side components.